frontend/src/Components/NewCharts/LineChart.js

```diff
@@ -96,15 +99,19 @@ export default function LineChart({
 
   return (
     <>
-      <div ref={setReferenceElement} />
-      <Tooltip
-        ref={setPopperElement}
-        style={{ ...styles.popper, width: '300px' }}
-        {...attributes.popper}
-      >
-        {tooltipText}
-      </Tooltip>
-      <Line options={options} data={data} />
+      {displayStopPurposeTooltips && (
+        <>
+          <div ref={setReferenceElement} />
+          <Tooltip
+            ref={setPopperElement}
+            style={{ ...styles.popper, width: '300px' }}
+            {...attributes.popper}
+          >
+            {tooltipText}
+          </Tooltip>
+        </>
+      )}
+      <Line options={options} data={data} redraw={redraw} datasetIdKey={title} />
```

Comment:
The `redraw` flag adjusts the widths of the adjacent graphs, increasing/decreasing depending on the selected graphs the user wants to see.